File: backend/src/main/java/com/springbootjwtpostgres/backend/orderdetail/OrderDetailSearchCriteria.java
package com.springbootjwtpostgres.backend.orderdetail;

import com.springbootjwtpostgres.backend.order.Order;
import lombok.Data;

import java.util.Date;

@Data
public class OrderDetailSearchCriteria {
    private int quantityFrom;
    private int quantityTo;
    private double totalPricePerProductFrom;
    private double totalPricePerProductTo;
    private Date createdAtFrom;
    private Date createdAtTo;
    private Long orderId;
    private Long productId;
}

File: frontend/src/app/models/orderdetail/OrderDetailSearchCriteria.ts
export class OrderDetailSearchCriteria {
  quantityFrom: number;
  quantityTo: number;
  totalPricePerProductFrom: number;
  totalPricePerProductTo: number;
  createdAtFrom: Date;
  createdAtTo: Date;
  orderId: number;
  productId: number;

  constructor(
    quantityFrom: number,
    quantityTo: number,
    totalPricePerProductFrom: number,
    totalPricePerProductTo: number,
    createdAtFrom: Date,
    createdAtTo: Date,
    orderId: number,
    productId: number
  ) {
    this.quantityFrom = quantityFrom;
    this.quantityTo = quantityTo;
    this.totalPricePerProductFrom = totalPricePerProductFrom;
    this.totalPricePerProductTo = totalPricePerProductTo;
    this.createdAtFrom = createdAtFrom;
    this.createdAtTo = createdAtTo;
    this.orderId = orderId;
    this.productId = productId;
  }
}
